codebase/src/components/shapes.ts

```diff
@@ -0,0 +1,52 @@
+import type { GEdge, GNode } from "@/graph/types";
+
+type GraphObject = {
+	data: GEdge | GNode,
+	isIntersecting: (data: GEdge | GNode, mouseX: number, mouseY: number) => boolean,
```

Comment:
is intersecting is "shapes" job. as laid out in the shape hit box definition. GraphObject should just pair a node or edge with its shape.